packages/2d/src/lib/components/Latex.ts

```diff
@@ -246,4 +262,175 @@ export class Latex extends SVGNode {
     yield* this.svg(newSVG, time, timingFunction);
     this.svg(this.latexSVG);
   }
+
+  public getPaths() {
+    return this.childAs(0)
+      ?.children()
+      .flatMap(part =>
+        part.children().length ? (part.children() as Path[]) : part,
+      ) as Path[];
+  }
+
+  public *write(time: number, timingFunction?: TimingFunction) {
+    const paths = this.getPaths();
+    const duration = time / paths.length;
+    const animations: ThreadGenerator[] = [];
+
+    for (let i = 0; i < paths.length; i++) {
+      paths[i].fill(null).stroke(this.color).lineWidth(40).start(0).end(0);
+      animations.push(
+        delay(
+          (i * duration) / FACTOR,
+          paths[i].end(1, duration, timingFunction),
+        ),
+        delay(
+          ((i + 1) * duration) / FACTOR,
+          paths[i].fill(this.color, duration, timingFunction),
+        ),
+        delay(
+          ((i + 1) * duration) / FACTOR,
+          paths[i].lineWidth(0, duration, timingFunction),
+        ),
+      );
+    }
+
+    yield* all(...animations);
+  }
+
+  public *unwrite(time: number, timingFunction?: TimingFunction) {
+    const paths = this.getPaths();
+    const duration = time / paths.length;
+    const animations: ThreadGenerator[] = [];
+
+    for (let i = 0; i < paths.length; i++) {
+      paths[i].lineWidth(0).stroke(this.color).start(0);
+      animations.push(
+        delay(
+          (i * duration) / FACTOR,
+          paths[i].fill(null, duration, timingFunction),
+        ),
+        delay(
+          (i * duration) / FACTOR,
+          paths[i].lineWidth(40, duration, timingFunction),
+        ),
+        delay(
+          ((i + 1) * duration) / FACTOR,
+          paths[i].start(1, duration, timingFunction),
+        ),
+      );
+    }
+
+    yield* all(...animations);
+  }
+
+  public *edit(tex: string, time: number) {
+    yield* this.tex(math(tex), time);
+  }
+
+  public *morph(ptx: Latex, time: number) {
+    const paths = this.getPaths();
+    const targets = ptx.getPaths();
+    const animations: ThreadGenerator[] = [];
+
+    for (let i = 0; i < paths.length; i++) {
+      paths[i].stroke(this.fill());
+
+      animations.push(
+        paths[i].fill(null, time * BUFFER),
+        paths[i].lineWidth(40, time * BUFFER),
+        delay(
+          time * BUFFER,
+          paths[i].data(targets[i].data(), time * (1 - 2 * BUFFER)),
+        ),
+        delay(
+          time * BUFFER,
+          paths[i].position(targets[i].position(), time * (1 - 2 * BUFFER)),
+        ),
+        delay(
+          time * BUFFER,
+          paths[i].scale(targets[i].scale(), time * (1 - 2 * BUFFER)),
+        ),
+        delay(time * (1 - BUFFER), paths[i].fill(paths[i].stroke(), time / 2)),
+        delay(time * (1 - BUFFER), paths[i].lineWidth(0, time / 2)),
+      );
+    }
+
+    yield* all(...animations);
+  }
+
+  public *map(ptx: Latex, mapping: number[][], time: number) {
```

Comment:
I'd like for this function to be a two-phase function, where the first phase is determining where everything should go, and the second applies it. Right now, it's pushing animations as it's also doing the analysis which makes it a little hard to follow. Ideally we'd be able to share the path-morphing parts between `map` and `morph`, maybe with a private `morphPath` function.

Reply:
- [ ] Restructure into a two-phase function
- [ ] Ideally, `yield` animations rather than building up a big list of them.